qiskit_optimization/translators/ising.py

```diff
@@ -0,0 +1,255 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Translator between an Ising Hamiltonian and a quadratic program"""
+
+from typing import Dict, Tuple, Union
+
+from numpy import zeros
+from qiskit.opflow import I, ListOp, OperatorBase, PauliOp, PauliSumOp, SummedOp
+from qiskit.quantum_info import Pauli
+
+from qiskit_optimization.exceptions import QiskitOptimizationError
+from qiskit_optimization.problems.quadratic_program import QuadraticProgram
+
+
+def to_ising(quad_prog: QuadraticProgram) -> Tuple[OperatorBase, float]:
+    """Return the Ising Hamiltonian of this problem.
+
+    Variables are mapped to qubits in the same order, i.e.,
+    i-th variable is mapped to i-th qubit.
+    See https://github.com/Qiskit/qiskit-terra/issues/1148 for details.
+
+    Returns:
+        qubit_op: The qubit operator for the problem
+        offset: The constant value in the Ising Hamiltonian.
+
+    Raises:
+        QiskitOptimizationError: If a variable type is not binary.
+        QiskitOptimizationError: If constraints exist in the problem.
+    """
+    # if problem has variables that are not binary, raise an error
+    if quad_prog.get_num_vars() > quad_prog.get_num_binary_vars():
+        raise QiskitOptimizationError(
+            "The type of variable must be a binary variable. "
```

Comment:
How above `The type of all variables must be binary. ` so its plural as otherwise it seems a little odd

Reply:
Thanks. I replace the message.